packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -86,15 +90,103 @@ const VeniaAdapter = props => {
         return null;
     }
 
+    let storeCodeRouteHandler = '';
+    const browserRouterProps = {};
+    if (process.env.USE_STORE_CODE_IN_URL === 'true') {
+        const storeCode = storage.getItem('store_view_code') || STORE_VIEW_CODE;
+        browserRouterProps.basename = `/${storeCode}`;
+        const storeCodes = [];
+        const storeCurrencies = {};
+        if (Array.isArray(AVAILABLE_STORE_VIEWS)) {
+            AVAILABLE_STORE_VIEWS.forEach(store => {
+                storeCodes.push(store.code);
+                storeCurrencies[store.code] =
+                    store.default_display_currency_code;
+            });
+        }
+
+        /**
+         * The current store code won't be matched as it's included as the basename, if we match another store
+         * we need to change the current store.
+         */
+        storeCodeRouteHandler = (
+            <Switch>
+                <Route path={`/:storeCode(${storeCodes.join('|')})?`}>
+                    {({ match }) => {
```

Comment:
Now that we have hooks, we should avoid function-as-a-child wherever possible. Not only are components more readable and performant, they can call hooks—whereas function children can't.

It's even more relevant in this case because the function child here is performing side effects, such as writing to local storage and reloading `window.location`. These effects [must be encapsulated in `useEffect`](https://reactjs.org/docs/hooks-reference.html#useeffect):

> Mutations, subscriptions, timers, logging, and other side effects are not allowed inside the main body of a function component (referred to as React’s render phase). Doing so will lead to confusing bugs and inconsistencies in the UI. 

Consider extracting this logic to a real component.

```es6
import { useRouteMatch } from "react-router-dom"

const StoreCodeRoute => () => {
  const pattern = `/:storeCode(${storeCodes.join()})?`
  const match = useRouteMatch(pattern)

  // ...

  useEffect(() => {
    // storage.setItem()
  }, [])
}
```

Reply:
Updated.